crates/cargo-test-support/src/registry.rs

```diff
@@ -60,6 +60,17 @@ pub fn alt_api_url() -> Url {
     Url::from_file_path(alt_api_path()).ok().unwrap()
 }
 
+fn generate_path(name: &str) -> PathBuf {
+    paths::root().join(name)
+}
+fn generate_url(name: &str) -> Url {
+    Url::from_file_path(generate_path(name)).ok().unwrap()
+}
+fn generate_dl_url(name: &str) -> String {
+    let base = Url::from_file_path(generate_path(name)).ok().unwrap();
+    format!("{}/{{crate}}/{{version}}/{{crate}}-{{version}}.crate", base)
+}
```

Comment:
Can all the other functions (the `alt_*` functions) be changed to call these new functions instead of duplicating the code?

Reply:
Yes, I fixed it like that. I also corrected `registry_path` function etc., but is this OK?